parallel_spectral_fitting/parallel_spectral_fitting.ipynb

```diff
@@ -0,0 +1,980 @@
+    "The magic of dask is using the [dask.array.map_blocks](https://docs.dask.org/en/latest/array-api.html#dask.array.map_blocks) function. We will write an additional function to handle the input and output to complete the fitting with the `map_blocks` function."
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def dask_parallel_spectrum_fitter(cube, guesses=None, compute=False):\n",
+    "    \"\"\"\n",
+    "    Fit all unmasked spectra in a cube.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    cube : `~spectral_cube.DaskSpectralCube`\n",
+    "        The spectral cube to fit.\n",
+    "    guesses : list of numpy.ndarray\n",
+    "        3-element list containing the (1) peak amplitude, (2) mean velocity, and\n",
+    "        (3) line width initial guesses.\n",
+    "    compute : bool, optional\n",
+    "        Compute the `map_blocks` computation for fitting. Otherwise, returns the\n",
+    "        uncomputed dask array.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    # Extract the underylying dask array in the cube\n",
+    "    data = cube._get_filled_data(fill=np.nan)\n",
+    "\n",
+    "    # Since we are manipulating only the dask array, we need to give the spectrum axis\n",
+    "    # separately\n",
+    "    spectrum_axis = cube.spectral_axis.value\n",
+    "\n",
+    "    # If no guesses are given, compute the simplest guesses to use\n",
+    "    if guesses is None:\n",
+    "        guesses = [cube.max(axis=0).value,\n",
+    "                   cube.moment1().value,\n",
+    "                   cube.linewidth_sigma().value]\n",
+    "\n",
+    "    # Set output chunk sizes.\n",
+    "    spat_chunks = tuple([chunk[0] for chunk in data.chunks[1:]])\n",
+    "\n",
+    "    out = cube.apply_function_parallel_spectral(spectrum_fit_gaussian,\n",
+    "                                                accepts_chunks=True,\n",
+    "                                                spectrum_axis=spectrum_axis,\n",
+    "                                                debug=False,\n",
+    "                                                guesses=guesses)\n",
+    "\n",
+    "    # # Use the dask schedular settings set for `cube`\n",
+    "    # with dask.config.set(**cube._scheduler_kwargs):\n",
+    "    #     # map_blocks applies the fitting function to all chunks\n",
+    "    #     out = da.map_blocks(spectrum_fit_gaussian,\n",
+    "    #                         data,\n",
+    "    #                         spectrum_axis=spectrum_axis,\n",
+    "    #                         debug=False,\n",
+    "    #                         drop_axis=[0],  # Tell dask we're not returning the spectral axis\n",
+    "    #                         chunks=(3,) + spat_chunks, # Instead, the 0th axis will have the 3 fit parameters\n",
+    "    #                         guesses=guesses)\n",
```

Comment:
what's the intent with this?  showing how you _would_ do this, or is this to be removed b/c it's in the func above

Reply:
The latter. This call is just a currently working example

Reply:
Removed now. With #714, we should only need the fitting function, not the map blocks wrapper.